calour/io.py

```diff
@@ -80,13 +84,50 @@ def _read_biom(fp, transpose=True):
     return sid, fid, data, feature_md
 
 
-def _read_qiime2(fp, transpose=True):
-    '''Read in a qiime2 .qza biom table file.
+def _file_from_zip(tempdir, data_file, internal_data):
+    '''extract the data file from a regular/qza filename into the tempdir, and return the filename
+
+    Parameters
+    ----------
+    tmpdir: str
+        name of the directory to extract the zip into
+    data_file: str
+        original name of the file (could be '.qza' or not)
+    internale_data: str
+        the internal qiime2 qza file name (i.e. 'data/feature-table.biom' for biom table etc.)
+
+    Returns
+    -------
+    str: name of data file inside the tempdir
+    '''
+    if not zipfile.is_zipfile(data_file):
+        raise ValueError('the qiime2 file %s is not a valid zip file. Is it a qiime2 artifact (.qza) file?' % data_file)
+    fl = zipfile.ZipFile(data_file)
```

Comment:
this should use `with`, like open a file object

Reply:
fixed